Directory.Build.props

```diff
@@ -1,4 +1,4 @@
-<Project>
+<Project TreatAsLocalProperty="RepoRoot">
```

Comment:
Why do we need `TreatAsLocalProperty` here. Usually we try to avoid setting this as it lead to several errors in the past in corefx. RepoRoot is always constructed with a trailing slash in [Arcade](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L23) and `MSBuildThisFileDirectory` also guarantees to contain the trailing slash.

Reply:
We need `TreatAsLocalProperty` here because in some cases (specifically when using the `LibrariesConfiguration` property via the new switch), we were getting a no-trailing-slash RepoRoot passed down that I couldn't trail down the cause of other than allowing us to reset it in Arcade. This was the only location where we were setting `RepoRoot`, so I felt this was the right location for the fix.